### PR TITLE
Unify `Example` and `Outcome`

### LINKED ISSUE #61: Unify `Example` and `Outcome`
This will need to be done on `staging` as it's an API break.

### PR DESCRIPTION
**Pull request by re-xyr**
Fixes #61 & blocked by #58

### CODE REVIEW

File: tasty-plutus/src/Test/Tasty/Plutus/TestData.hs [hunk 1]
```diff
@@ -69,25 +69,25 @@ data TestData (p :: Purpose) where
 {- | Describes whether a case is good (i.e. should pass) or bad (i.e. should
```
Comment (kozross) [inline tasty-plutus/src/Test/Tasty/Plutus/TestData.hs:69]: The documentation should be adjusted to take into account `Outcome`'s new unified role.

**Review by kozross — CHANGES_REQUESTED** [review]
We need to do two more things:

1. Add a changelog entry about the unification; and
2. Bump to an even higher major version, since 4.0 is already taken by the Plutus bump.

You can also modify the changelog to have the 'Unreleased' section be '5.0' instead, along with all the changes done so far.

File: tasty-plutus/src/Test/Tasty/Plutus/TestData.hs (hunk 1 above)
Comment (re-xyr) [reply tasty-plutus/src/Test/Tasty/Plutus/TestData.hs:69]: For the public API, the `Outcome` type is still only used for QuickCheck-based tests. So I don't think we'll change this part?

Comment (re-xyr) [reply tasty-plutus/src/Test/Tasty/Plutus/TestData.hs]: I still tried to change the comments; overall I'm not sure if I should mention internal usages in user-facing docs.

**Comment by re-xyr** [comment]
Just found that there has been a 4.1 release already - so I moved the changelog entry to the 'Unreleased' section.

**Review by kozross — APPROVED** [review]

**Comment by kozross** [comment]
Great - nicely done!

**Pull request merged**